dom/src/main/scala/fs2/dom/ResizeObserver.scala

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2022 Arman Bilge
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package fs2.dom
+
+import scalajs.js
+import org.scalajs.dom
+import cats.effect.kernel.Async
+import cats.effect.std.Dispatcher
+import cats.effect.kernel.Resource
+
+trait ResizeObserver[F[_]] {
+
+  def observe(target: dom.Element): F[Unit]
+
+  def observe(target: dom.Element, options: dom.ResizeObserverOptions): F[Unit]
+
+  def unobserve(target: dom.Element): F[Unit]
+
+  def disconnect(): F[Unit]
+
+}
+
+object ResizeObserver {
+
+  def apply[F[_]](
+      callback: (js.Array[dom.ResizeObserverEntry], dom.ResizeObserver) => F[Unit]
```

Comment:
This is going to be fun 😂 
```suggestion
      callback: (List[dom.ResizeObserverEntry], ResizeObserver[F]) => F[Unit]
```